cpp/core/jni/JniWrapper.cc

```diff
@@ -770,6 +779,87 @@ JNIEXPORT jlong JNICALL Java_io_glutenproject_vectorized_ShuffleSplitterJniWrapp
   JNI_METHOD_END(-1L)
 }
 
+JNIEXPORT jlong JNICALL Java_io_glutenproject_vectorized_ShuffleSplitterJniWrapper_nativeMakeForCeleborn(
+    JNIEnv* env,
+    jobject,
+    jstring partitioning_name_jstr,
+    jint num_partitions,
+    jlong offheap_per_task,
+    jint buffer_size,
+    jstring compression_type_jstr,
+    jint batch_compress_threshold,
+    jint push_buffer_max_size,
+    jobject celeborn_partition_pusher,
+    jlong allocator_id,
+    jlong firstBatchHandle) {
+  JNI_METHOD_START
+  if (partitioning_name_jstr == NULL) {
+    gluten::JniThrow(std::string("Short partitioning name can't be null"));
+    return 0;
+  }
+  auto partitioning_name_c = env->GetStringUTFChars(partitioning_name_jstr, JNI_FALSE);
+  auto partitioning_name = std::string(partitioning_name_c);
+  env->ReleaseStringUTFChars(partitioning_name_jstr, partitioning_name_c);
+  auto splitOptions = SplitOptions::Defaults();
+  splitOptions.buffered_write = true;
+  splitOptions.is_celeborn = true;
+  if (buffer_size > 0) {
+    splitOptions.buffer_size = buffer_size;
+  }
+  if (push_buffer_max_size > 0) {
+    splitOptions.push_buffer_max_size = push_buffer_max_size;
+  }
+  splitOptions.offheap_per_task = offheap_per_task;
+  if (compression_type_jstr != NULL) {
+    auto compression_type_result = GetCompressionType(env, compression_type_jstr);
+    if (compression_type_result.status().ok()) {
+      splitOptions.compression_type = compression_type_result.MoveValueUnsafe();
+    }
+  }
+
+  auto* allocator = reinterpret_cast<MemoryAllocator*>(allocator_id);
+  if (allocator == nullptr) {
+    gluten::JniThrow("Memory pool does not exist or has been closed");
+  }
+  splitOptions.memory_pool = AsWrappedArrowMemoryPool(allocator);
+  jclass cls = env->FindClass("java/lang/Thread");
+  jmethodID mid = env->GetStaticMethodID(cls, "currentThread", "()Ljava/lang/Thread;");
+  jobject thread = env->CallStaticObjectMethod(cls, mid);
+  if (thread == NULL) {
+    std::cout << "Thread.currentThread() return NULL" << std::endl;
+  } else {
+    jmethodID mid_getid = GetMethodIDOrError(env, cls, "getId", "()J");
+    jlong sid = env->CallLongMethod(thread, mid_getid);
+    splitOptions.thread_id = (int64_t)sid;
+  }
+  jclass tc_cls = env->FindClass("org/apache/spark/TaskContext");
+  jmethodID get_tc_mid = env->GetStaticMethodID(tc_cls, "get", "()Lorg/apache/spark/TaskContext;");
+  jobject tc_obj = env->CallStaticObjectMethod(tc_cls, get_tc_mid);
+  if (tc_obj == NULL) {
```

Comment:
This revoke java method looks complicated, why not send it as function arg?

Reply:
@jinchengchenghh This part reuses the code of nativeMake. I think it is feasible to replace revoke with parameter passing. I checked this part was developed by @rui-mo at the time. 
May I ask what was the reason for using revoke in this part at that time?  @rui-mo

Reply:
@jinchengchenghh @rui-mo nativeMakeForCeleborn has been optimized, if there is a need in the future, I will submit a new pr to optimize nativeMake.